repos/fdbt-site/src/pages/api/deletePassenger.ts

```diff
@@ -25,6 +28,30 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
             throw new Error('This individual cannot be deleted because it is in use in a group.');
         }
 
+        const products = await getAllProductsByNoc(nationalOperatorCode)
+        const matchingJsonLinks = products.map((product) => product.matchingJsonLink);
+        const tickets = await Promise.all(
+            matchingJsonLinks?.map(async (link) => {
+                return await getProductsMatchingJson(link);
+            }),
+        );
+
+        const productsUsingPassengerType = tickets?.filter(t => t?.passengerType?.id === id)
+        const productNames = productsUsingPassengerType?.map(ticket=> ticket.products[0] ? (ticket.products[0].productName || 'missing'): "")
+        if(productsUsingPassengerType && productsUsingPassengerType.length > 0){
+            const passengerDetails : any = await getPassengerTypeById(id, nationalOperatorCode)
```

Comment:
const passengerType  =  await getPassengerTypeById(id, nationalOperatorCode)

Reply:
how would i prevent the typescript error here it will throw one without it?